SignalServiceKit/src/Messages/InvalidKeyMessages/TSInvalidIdentityKeyReceivingErrorMessage.m

```diff
@@ -66,11 +66,11 @@ - (nullable instancetype)initForUnknownIdentityKeyWithTimestamp:(uint64_t)timest
     return self;
 }
 
-- (nullable SSKEnvelope *)envelope
+- (nullable SSKProtoEnvelope *)envelope
 {
     if (!_envelope) {
         NSError *error;
-        SSKEnvelope *_Nullable envelope = [[SSKEnvelope alloc] initWithSerializedData:self.envelopeData error:&error];
+        SSKProtoEnvelope *_Nullable envelope = [SSKProtoEnvelope parseData:self.envelopeData error:&error];
```

Comment:
One departure from the `SSKEnvelope` is that I'm doing the parsing in a factory method rather than an initializer.

Reply:
I suppose it's mostly a matter of taste.  I prefer initializers that are "dumb" and don't do a bunch of work.